pkg/controller/studyjob/studyjob_controller.go

```diff
@@ -393,46 +394,55 @@ func (r *ReconcileStudyJobController) updateWorker(c katibapi.ManagerClient, ins
 	return update, nil
 }
 
-func (r *ReconcileStudyJobController) getJobWorkerStatus(w *katibv1alpha1.WorkerCondition, ns string) WorkerStatus {
-	runtimejob := createWorkerJobObj(w.Kind)
-	nname := types.NamespacedName{Namespace: ns, Name: w.WorkerID}
-	joberr := r.Client.Get(context.TODO(), nname, runtimejob)
-	if joberr != nil {
-		return WorkerStatus{}
-	}
+func (r *ReconcileStudyJobController) getJobWorkerStatus(ns string, wid string, wkind *schema.GroupVersionKind) WorkerStatus {
+	nname := types.NamespacedName{Namespace: ns, Name: wid}
 	var state katibapi.State = katibapi.State_RUNNING
 	var cpTime *metav1.Time
-	switch w.Kind {
+	switch wkind.Kind {
+
 	case DefaultJobWorker:
-		job := runtimejob.(*batchv1.Job)
+		var job batchv1.Job
+		if err := r.Client.Get(context.TODO(), nname, &job); err != nil {
+			log.Printf("Client Get error %v for %v", err, nname)
+			return WorkerStatus{}
+		}
 		if job.Status.Active == 0 && job.Status.Succeeded > 0 {
```

Comment:
It seems like we should be able to use the similar logic here to check for job completion: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/batch/v1/generated.proto#L158

Reply:
Please see above comment